internal/batches/service/remote.go

```diff
@@ -98,13 +113,122 @@ func (svc *Service) CreateBatchSpecFromRaw(
 		"allowIgnored":     allowIgnored,
 		"allowUnsupported": allowUnsupported,
 		"noCache":          noCache,
+		"batchChange":      batchChange,
 	}).Do(ctx, &resp); err != nil || !ok {
 		return "", err
 	}
 
 	return resp.CreateBatchSpecFromRaw.ID, nil
 }
 
+// UploadMounts uploads file mounts to the server.
+func (svc *Service) UploadMounts(workingDir string, batchSpecID string, steps []batches.Step) error {
+	if err := svc.areServerSideBatchChangesSupported(); err != nil {
+		return err
+	}
+
+	body := &bytes.Buffer{}
+	w := multipart.NewWriter(body)
+
+	var count int
+	for _, step := range steps {
+		for _, mount := range step.Mount {
+			total, err := handlePath(w, workingDir, mount.Path, count)
+			if err != nil {
+				return err
+			}
+			count += total
+		}
+	}
+
+	if err := w.WriteField("count", strconv.Itoa(count)); err != nil {
+		return err
+	}
+
+	// Honestly, the most import thing to do. This adds the closing boundary to the request.
+	if err := w.Close(); err != nil {
+		return err
+	}
+
+	request, err := svc.client.NewHTTPRequest(context.Background(), http.MethodPost, fmt.Sprintf(".api/batches/mount/%s", batchSpecID), body)
```

Comment:
I think this should take a non-background context so we can handle cancellation right